rfcs/0147-vm-syscalls-3/0147-vm-syscalls-3.md

```diff
@@ -0,0 +1,188 @@
+---
+Number: "0147"
+Category: Standards Track
+Status: Draft
+Author: Xu Jiandong <lynndon@gmail.com>, Dingwei Zhang <zhangsoledad@gmail.com>
+Created: 2023-04-17
+---
+
+# VM Syscalls 3
+
+## Abstract
+
+This document describes the addition of the syscalls during the CKB2023. This update significantly enhances the flexibility of CKB Script.
+
+The following three syscalls are added:
+
+- [Spawn]
+- [Get Memory Limit]
+- [Set Content]
+- [Load Extension]
+
+### Spawn
+[Spawn]: #spawn
+
+The syscall Spawn is the core part of this update. The *Spawn* and the latter two syscalls: *Get Memory Limit* and *Set Content* together, implement a way to call another CKB Script in a CKB Script. Unlike the *Exec*[1](../0034-vm-syscalls-2/0034-vm-syscalls-2.md) syscall, *Spawn* saves the execution context of the current script, like [posix_spawn](https://man7.org/linux/man-pages/man3/posix_spawn.3.html), the parent script blocks until the child script ends.
+
+```c
+int ckb_spawn(uint64_t memory_limit, size_t index, size_t source,
```

Comment:
Note that we have updated the signature of spawn.  Please update it from internal docs.

Reply:
@mohanson updated